package/src/index.types.ts

```diff
@@ -0,0 +1,72 @@
+import { z } from 'zod';
+// Constants
+import { MAX_HITS_PER_PAGE, MAX_PAGES } from './constants.js';
+
+/**
+ * Input validation with zod
+ * Succesful validation returns a typed object
+ */
+
+const SearchParams = z.object({
+  query: z.string(),
+  // Optional pagination params
+  page: z.number().gte(0).lte(MAX_PAGES).optional(),
+  hitsPerPage: z.number().gte(1).lte(MAX_HITS_PER_PAGE).optional(),
+  // Unused params
+  facets: z.array(z.string()).optional(),
+  highlightPostTag: z.string().optional(),
+  highlightPreTag: z.string().optional(),
+  tagFilters: z.string().optional(),
+});
+
+export const Json = z.object({
+  params: SearchParams,
+  indexName: z.string(),
+});
+
+/**
+ * Types
+ */
+
+export interface GenericReq {
```

Comment:
This would bo ok if the handler was just for express. But, for it to work with nextjs and more, the type needs to be less strict.

Here's a nextjs example with express types.

```ts
// api/search.ts
import { searchHandler } from 'postgres-searchbox';
import type { NextApiRequest, NextApiResponse } from 'next';

export default (req: NextApiRequest, res: NextApiResponse) =>
  searchHandler(req, res);
```

Gives TS error.

Argument of type 'NextApiRequest' is not assignable to parameter of type 'Request<ParamsDictionary, any, any, ParsedQs, Record<string, any>>'.
  Type 'NextApiRequest' is missing the following properties from type 'Request<ParamsDictionary, any, any, ParsedQs, Record<string, any>>': get, header, accepts, acceptsCharsets, and 23 more.

There could be a better way, but this works ok for now.

Reply:
Good point, thx.